### PR TITLE
KFSPTS-31686 add confirmation message when SSC creates DV for contract

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/module/purap/CUPurapKeyConstants.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class CUPurapKeyConstants {
 
     public static final String ATTACHMENT_QUESTION_CONFIRM_CHANGE = "attachment.message.confirm.change";
     
-    public static final String MESSAGE_IWNT_CONFIRM_CREATE_REQ = "message.createReq.confirmationMessage";
+    public static final String MESSAGE_IWNT_CONFIRM_CREATE_REQ_OR_DV = "message.createReqOrDV.confirmationMessage";
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/module/purap/CUPurapKeyConstants.java:31]: Should the "V" in "message.createReqOrDV.confirmationMessage" be lower case for consistent camel casing?

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java [hunk 2]
```diff
@@ -407,15 +408,53 @@ && isInOrgHierarchyNode(getDocument()) && KRADConstants.YES_INDICATOR_VALUE.equa
...
-        ExtraButton clearButton = new ExtraButton();
-        clearButton.setExtraButtonProperty("methodToCall.createDV");
-        clearButton.setExtraButtonSource("${" + KFSConstants.EXTERNALIZABLE_IMAGES_URL_KEY + "}buttonsmall_create_DV.gif");
-        clearButton.setExtraButtonAltText("Create DV");
-        clearButton.setExtraButtonOnclick("window.open('" + ConfigContext.getCurrentContextConfig().getProperty(KFSConstants.APPLICATION_URL_KEY)
-                + "/financialDisbursementVoucher.do?methodToCall=createDVFromIWantDoc&docId=" + getDocument().getDocumentNumber()
-                + "');return false;");
-        clearButton.setExtraButtonParams("_blank");
-        return clearButton;
+        ExtraButton creatDVButton = new ExtraButton();
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java:411]: Misspelling in the local variable name. 
Also, I am not sure the camel casing is correct here "...DV..." but that is what is used elsewhere throughout the file so I do not believe we should change it at this time.

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java [hunk 3]
```diff
@@ -439,8 +478,11 @@ public boolean isDocEnroute() {
         return ObjectUtils.isNotNull(workflowDocument) && workflowDocument.isEnroute();
     }
     
-    public String getContractWarningMessage() {
-        return getConfigurationService().getPropertyValueAsString(CUPurapKeyConstants.MESSAGE_IWNT_CONFIRM_CREATE_REQ);
+    public String getContractWarningMessage(String docType) {
+        String message = getConfigurationService().getPropertyValueAsString(CUPurapKeyConstants.MESSAGE_IWNT_CONFIRM_CREATE_REQ_OR_DV);
+        String messageFormattedText = MessageFormat.format(message,
+                docType);
```
Comment (nkimble) [inline src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java:484]: Is there a reason why we need two local variables here? 
Meaning, is there a reason why this cannot be consolidated into :
`return MessageFormat.format(getConfigurationService().getPropertyValueAsString(CUPurapKeyConstants.MESSAGE_IWNT_CONFIRM_CREATE_REQ_OR_DV), docType);`

**Review by nkimble — COMMENTED** [review]
The changes look good. I did find a couple area that may need cleanup.

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java (hunk 2 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java]: updated

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java (hunk 3 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java]: updated

File: src/main/java/edu/cornell/kfs/module/purap/CUPurapKeyConstants.java (hunk 1 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/CUPurapKeyConstants.java:31]: Good question. I find it easier to read as is but I can change it.

**Review by nkimble — APPROVED** [review]
Changes look good. My concerns have been addressed. Approved for develop branch.

**Pull request merged**